### ISSUE TITLE
python 3.8 changes line to disable pylint warning

### ISSUE THREAD

**Issue by Ge0rgi0s** [PyCQA/pylint#3295]
In python 3.7 the following code silenced the pylint warning:
```python
@dataclass  # pylint: disable=too-many-instance-attributes
class Test:
    """
    I have too many attributes, but as a dataclass I need them.
    """

    a_1: int
    a_2: int
    a_3: int
    a_4: int
    a_5: int
    a_6: int
    a_7: int
    a_8: int
```
but starting with python 3.8 the following is expected
```python
@dataclass
class Test:  # pylint: disable=too-many-instance-attributes
    """
    I have too many attributes, but as a dataclass I need them.
    """

    a_1: int
    a_2: int
    a_3: int
    a_4: int
    a_5: int
    a_6: int
    a_7: int
    a_8: int
```
**Steps to reproduce**
Run the first code with pylint under python 3.8 and the second with pylint under python 3.7.

**Expected behaviour**
I find the behaviour under python 3.8 more intuitive, but would not like to change the python 3.7 code I already have.

**pylint --version output**
```
pylint 2.4.4
astroid 2.3.3
Python 3.7.5 (default, Nov 23 2019, 05:59:34) 
[GCC 8.3.0]
```
and
```
pylint 2.4.4
astroid 2.3.3
Python 3.8.0 (default, Oct 23 2019, 18:51:26) 
[GCC 9.2.0]
```

**Comment by PCManticore** [comment]
Thanks for the report. I believe the line numbers of decorators changed in 3.8 so addressing this might need a patch in `astroid`.

**Comment by DanielNoord** [comment]
This has been fixed due to change in `astroid`. `3.7` now also requires the format of `3.8`.
